src/main/java/org/gbif/ipt/config/ConfigManagerImpl.java

```diff
@@ -211,13 +211,52 @@ public void loadDataDirConfig() throws InvalidConfigException {
     LOG.info("Loading resource configurations ...");
     // default creator used to populate missing resource creator when loading resources
     User defaultCreator = (userManager.list(Role.Admin).isEmpty()) ? null : userManager.list(Role.Admin).get(0);
-    resourceManager.load(dataDir.dataFile(DataDir.RESOURCES_DIR), defaultCreator);
+    File resourcesDir = dataDir.dataFile(DataDir.RESOURCES_DIR);
+    checkResourcesDirAtStartup(resourcesDir);
+    resourceManager.load(resourcesDir, defaultCreator);
 
     // start publishing monitor
     LOG.info("Starting Publishing Monitor...");
     publishingMonitor.start();
   }
 
+  private void checkResourcesDirAtStartup(File resourcesDir) {
+    // READ access of /resources
+    if (resourcesDir != null && resourcesDir.exists() && !resourcesDir.canRead()) {
```

Comment:
We check resourceDir is not null and exists five times. Please consider checking that only once to avoid multiple checks

Reply:
Sorry, three times for resourcesDir and twice for resourceDir.

Reply:
Indeed. Just commited a factorization of the test